src/temperature_tele.rs

```diff
@@ -0,0 +1,47 @@
+//! # Thermostat_EEM temperature telemetry processing
+
+use serde::Serialize;
+
+/// Temperature telemetry struct.
+#[derive(Serialize, Copy, Clone, Default, Debug)]
+pub struct Temperature {
+    average: f32,
+    min: f32,
+    max: f32,
+}
+
+/// Temperature buffer for computing min/max/average of the last telemetry period.
+#[derive(Default, Copy, Clone, Debug)]
+pub struct TemperatureBuffer {
+    min: f32,
+    max: f32,
+    accumulator: f32,
+    counter: u32,
+}
+
+impl TemperatureBuffer {
+    /// Add a new temperature sample to the buffer. This will add it to the accumulator,
+    /// update min/max and increment the counter.
+    pub fn add(&mut self, temp: f32) {
+        self.counter += 1;
+        self.accumulator += temp;
```

Comment:
Does `f32` have enough dynamic range? I suspect it doesn't.
At least make a note that this is OK as long as either the ADC values are sufficiently noisy or the telemetry period is short enough. Otherwise you'll get cancellation effects (addition).

Reply:
I thought about it while implementing it but then forgot at the end. 
Very rough estimate: 25°C * 500 is already 5 digits, so any further sample addition will only be accurate to about 0.01 °C (about 7 digits total for 24 bit mantissa), right?

Reply:
Worst case is t_tele/t_sample*T_max*f32::eps ~ 10*1000*100*1e-7 K = 0.1 K. So yes, it's bad.

Reply:
And it's even worse for stddev when using the naive algo.